### ISSUE TITLE
Creating private gke on shared vpc failes

### ISSUE THREAD

**Issue by oricoralogix** [terraform-google-modules/terraform-google-kubernetes-engine#1680]
### TL;DR

I'm using version 26.1.1" of the module 
I have host project X and service project y.
Configured with the following params:
project_id                 = var.project_id
   name                       = var.cluster_name
   region                     = var.region
   zones                      = var.zones
   network                    = "${var.host_project_id}-vpc"
   subnetwork                 = "${var.host_project_id}-${var.region}-subnet-0"
   network_project_id         = var.host_project_id
   ip_range_pods              = "${var.host_project_id}-${var.region}-pods-0"
   ip_range_services          = "${var.host_project_id}-${var.region}-services-0"
   http_load_balancing        = false
   network_policy             = false
   horizontal_pod_autoscaling = true
   filestore_csi_driver       = false
   enable_private_endpoint    = true
   enable_private_nodes       = true
   master_ipv4_cidr_block     = "10.0.0.0/28"
   remove_default_node_pool   = true

When running apply error raised with the wrong config:
Error: Error creating NetworkPeeringRoutesConfig: googleapi: Error 404: The resource 'projects/<y-project>/global/networks/<x-project>-vpc' was not found, notFound 

### Expected behavior

Peering is created or looked for at project X which is the host project and not project Y which is the service project

### Observed behavior

Peering is created on the service project and not on the host

### Terraform Configuration

```hcl
project_id                 = var.project_id
   name                       = var.cluster_name
   region                     = var.region
   zones                      = var.zones
   network                    = "${var.host_project_id}-vpc"
   subnetwork                 = "${var.host_project_id}-${var.region}-subnet-0"
   network_project_id         = var.host_project_id
   ip_range_pods              = "${var.host_project_id}-${var.region}-pods-0"
   ip_range_services          = "${var.host_project_id}-${var.region}-services-0"
   http_load_balancing        = false
   network_policy             = false
   horizontal_pod_autoscaling = true
   filestore_csi_driver       = false
   enable_private_endpoint    = true
   enable_private_nodes       = true
   master_ipv4_cidr_block     = "10.0.0.0/28"
   remove_default_node_pool   = true
```

### Terraform Version

```sh
1.4.6
```

### Additional information

Am I doing something wrong or miss configured? 
Didn't find anyway yet to pre-configured the peering and give it as a variable to the module ...
